### ISSUE TITLE
Dev/Prod inconsistency for Ember.Object.create().hasOwnProperty('_super')

### ISSUE THREAD

**Issue by dschmidt** [emberjs/ember.js#12462]
Hey,

today I ran into an issue with our production builds being broken,  which was ultimately caused by minified builds behaving differently to non-minified builds:
`Ember.Object.create().hasOwnProperty('_super')` is `false` in non-minified builds, but `true` in minified ones.

I've actually verified that building in `production` environment with `minifyJS: { enabled: false }` isn't making `_super` an 'own property'. So I assume it's a minification issue and nothing else that might differ for production and development builds.

We stumbled upon this because we were doing `for in` on an `Ember.Object` and checking the properties with `hasOwnProperty`. (yes, it's fishy. no, i don't think it's a good idea. yes, i've changed the code already, but the inconsisteny *somewhere* remains).

To show this behavior I prepared a demo project here:
https://github.com/dschmidt/ember-dev-prod-inconsistency
(It's just a completely empty project ... but any of your projects should be good)

Just run `ember s` and `ember s --environment=production` and compare the outputs of `Ember.Object.create().hasOwnProperty('_super')`

BR,
Domme

**Comment by rwjblue** [comment]
Just confirmed that `hasOwnProperty('_super')` is `false` when running debug or prod builds without minification and `true` when running minified builds.

**Comment by rwjblue** [comment]
OK, so I have tracked down the issue.  

We check to see if a function contains `_super` / `.call` / `.apply` before attempting to super wrap a given method (this saves quite a bit of extra super wrapping for functions that do not need it). Unfortunately, a number of platforms that we support do not support calling `func.toString()` so we attempt to detect this and fall back to always super wrap everything mode.

We have roughly this code (from [here](https://github.com/emberjs/ember.js/blob/b4718218dbe5ffe7736c485a594248b20977c621/packages/ember-metal/lib/utils.js#L257-L271)) to detect if we can call `toString()` and get the original source:

```javascript
  let sourceAvailable = (function() {
    return this;
  }).toString().indexOf('return this;') > -1;
```

This works perfectly for development builds, but unfortunately not when minified.  Take a look at the minified source and you will see why:

```javascript
var e=function(){return this}.toString().indexOf("return this;")>-1;
```

Note that minifier has stripped the trailing semicolon from the function body we are attempting to check against, but sadly our `indexOf` check still contains the semicolon!

---

**tldr;** We super wrap every function in uglified builds.
